deepchecks/recommender/ranking.py

```diff
@@ -7,705 +7,423 @@
+                        recommendations: Union[List, np.array],
+                        k: Union[int, None] = None) -> float:
+    """
+    Compute the mean average Normalized Discounted Cumulative Gain (NDCG@k).
+
+    Parameters:
+    relevant_items (Union[List,np.array]): relevant items for each query/user.
+    recommendations (Union[List,np.array]): recommended items for each query/user.
+    k (Union[int, None]): The number of top items to consider.
     Returns:
-        (float): personalization
-
-    `Original <https://github.com/statisticianinstilettos/recmetrics/blob/master/recmetrics/metrics.py#L160>`_
+    float: Mean Average NDCG@k score.
     """
+    mean_ndcg = 0
+    num_items = relevant_items.shape[0]
 
-    n_predictions = len(recommendations)
+    # Compute the NDCG@k for each query/user and add to mean_ndcg
+    for item, recommendation in zip(relevant_items, recommendations):
+        mean_ndcg += ndcg_k(item, recommendation, k)
 
-    # map each ranked item to index
-    item2ix = {}
-    counter = 0
-    for prediction in recommendations:
-        for item in prediction:
-            if item not in item2ix:
-                item2ix[item] = counter
-                counter += 1
+    # Compute the mean NDCG@k and return
+    return mean_ndcg / num_items
 
-    n_items = len(item2ix.keys())
-
-    # create matrix of predicted x items
-    items_matrix = np.zeros((n_predictions, n_items))
-
-    for i, prediction in enumerate(recommendations):
-        for item in prediction:
-            item_ix = item2ix[item]
-            items_matrix[i][item_ix] = 1
-
-    similarity = cosine_similarity(X=items_matrix)
-    dim = similarity.shape[0]
-    personalization = (similarity.sum() - dim) / (dim * (dim - 1))
-
-    return 1 - personalization
-
-
-def novelty(recommendation: List, item_popularity: dict, num_users: int, k: int = 10):
+def popularity_based_novelty(recommendations: Union[List[list], np.array],
+                             items_popularity: Union[dict, None],
+                             num_users: int,
+                             k: int = 20) -> Tuple[float, float]:
     """
-    Calculates the capacity of the recommender system to generate novel
-    and unexpected results.
-
-    Args:
-        recommendation (List): A 1 x N array of items, where M is the number
-                               of predicted lists and N the number of recommended items
-        item_popularity (dict): A dict mapping each item in the recommendations to a popularity value.
-                                  Popular items have higher values.
-        num_users (int): The number of users
-        k (int): The number of items considered in each recommendation.
+    Calculate the popularity-based novelty metric for a list of recommendations.
+    Parameters:
+        recommendations (Union[List[list], np.array]): List of recommendations for each user.
+        items_popularity (Union[dict, None]): Item popularity scores.
```

Comment:
Should add a few more words on what popularity is, as in the previous implementation.

Reply:
Fixed.